### DISCUSSION TITLE
salesforce-alm package is deprecated

### DISCUSSION THREAD

**Discussion by svishalkumar06** [forcedotcom/cli#2792] [Q&A] [answered]
Hi,

Previously we were using sfdx-cli V7 Now we have migrated to @salesforce/cli V2 version.

So previously we were using a salesforce-alm and salesforcedx packages in sfdx-cli.

But In the V2 version of salesforce cli those salesforce-alm and salesforcedx packages are deprecated. So what is the replacement for those functionalities?

We were using the following functionality from the salesforce-alm package -

getting getMetadataTypeFromSourcePath from require('salesforce-alm/dist/lib/source/metadataTypeFactory')
getting metadataRegistry from require('salesforce-alm/dist/lib/source/metadataRegistry')
getting srcDevUtil from require('salesforce-alm/dist/lib/core/srcDevUtil')
===
getting almError from require("salesforce-alm/dist/lib/core/almError")
getting Messages from require("salesforce-alm/dist/lib/messages");
getting  HumanReporter, TapReporter, JUnitReporter, JsonReporter from 'salesforce-alm/dist/commands/force/apex/test/report';
getting logger from require("salesforce-alm/dist/lib/core/logApi");
getting StreamClient from require("salesforce-alm/dist/lib/core/status");
getting consts from require("salesforce-alm/dist/lib/core/constants");
getting messages from require("salesforce-alm/dist/lib/messages");
getting almError from require("salesforce-alm/dist/lib/core/almError");
getting reporter_1 from require("salesforce-alm/dist/lib/test/reporter");
getting TestResults from require("salesforce-alm/dist/lib/test/testResults");

For the replacement in @salesforce/cli We can use the SDR library:
https://github.com/forcedotcom/source-deploy-retrieve
and sfdx-core library:
https://github.com/forcedotcom/sfdx-core

But we do not have  1-1 equivalents replacement for our functionality. Please help us on finding the 1-1 equivalent for our functionalities

**Comment by mshanemc** [comment, marked as answer]
It's important to keep in mind that a CLI or CLI plugin has its input/output (command flags and return types) as an API.  

All the "helper" code inside that's not exported is not something you should created dependencies on.  **It's not a public API**, even if node lets you see/use the code.  We don't make any promises to maintain the shape of those functions or keep 1-1 equivalents around permanently.

our list of plugins/libraries are here: https://github.com/salesforcecli/status

For example, you'll find those apex test reporters here: https://github.com/forcedotcom/salesforcedx-apex/tree/main/src/reporters.

Github search is your friend here.  Looking in `forcedotcom` and `salesforcecli` repos is your best bet.
